source/winUser.py

```diff
@@ -612,3 +616,55 @@ def SendInput(inputs):
 	n = len(inputs)
 	arr = (Input * n)(*inputs)
 	user32.SendInput(n, arr, sizeof(Input))
+
+# Windows Clipboard format constants
+CF_UNICODETEXT = 13
+
+@contextlib.contextmanager
+def openClipboard(hwndOwner=None):
+	"""
+	A context manager version of OpenClipboard from user32. 
+	Use as the expression of a 'with' statement, and CloseClipboard will automatically be called at the end. 
+	"""
+	if not windll.user32.OpenClipboard(hwndOwner):
+		raise RuntimeError("OpenClipboard failed")
+	try:
+		yield
+	finally:
+		windll.user32.CloseClipboard()
+
+def emptyClipboard():
+	windll.user32.EmptyClipboard()
+
+def getClipboardData(format):
+	# We only support unicode text for now
+	if format!=CF_UNICODETEXT:
+		raise ValueError("Unsupported format")
+	# Fetch the data from the clipboard as a global memory handle
+	h=windll.user32.GetClipboardData(format)
+	if h:
+		# Lock the global memory  while we fetch the unicode string
+		# But make sure not to free the memory accidentally -- it is not ours
+		h=winKernel.HGLOBAL(h,autoFree=False)
+		with h.lock() as addr:
+			if addr:
+				# Read the string from the local memory address
+				return wstring_at(addr)
+
+def setClipboardData(format,data):
+	# For now only unicode is a supported format
+	if format!=CF_UNICODETEXT:
+		raise ValueError("Unsupported format")
+	text=unicode(data)
+	# Allocate global memory
+	h=winKernel.HGLOBAL.alloc(winKernel.GMEM_MOVEABLE,(len(text)+1)*2)
+	# Acquire a lock to the global memory receiving a local memory address
+	with h.lock() as addr:

```

Comment:
Also here, what happens `if not addr`?

Reply:
I don't believe addr can ever be 0 here as GlobalAlloc would have failed and h would have been 0.